pkg/apis/operations/v1alpha1/type.go

```diff
@@ -91,67 +89,42 @@ type NodeUpgradeJobSpec struct {
 	// The default Concurrency value is 1.
 	// +optional
 	Concurrency int32 `json:"concurrency,omitempty"`
-}
-
-// UpgradeResult describe the result status of upgrade operation on edge nodes.
-// +kubebuilder:validation:Enum=upgrade_success;upgrade_failed_rollback_success;upgrade_failed_rollback_failed
-type UpgradeResult string
-
-// upgrade operation status
-const (
-	UpgradeSuccess               UpgradeResult = "upgrade_success"
-	UpgradeFailedRollbackSuccess UpgradeResult = "upgrade_failed_rollback_success"
-	UpgradeFailedRollbackFailed  UpgradeResult = "upgrade_failed_rollback_failed"
-)
 
-// UpgradeState describe the UpgradeState of upgrade operation on edge nodes.
-// +kubebuilder:validation:Enum=upgrading;completed
-type UpgradeState string
+	// CheckItems specifies the items need to be checked before the task is executed.
+	// The default CheckItems value is nil.
+	// +optional
+	CheckItems []string `json:"checkItems,omitempty"`
 
-// Valid values of UpgradeState
-const (
-	InitialValue UpgradeState = ""
-	Upgrading    UpgradeState = "upgrading"
-	Completed    UpgradeState = "completed"
-)
+	FailureTolerate string `json:"failureTolerate,omitempty"`
+}
 
 // NodeUpgradeJobStatus stores the status of NodeUpgradeJob.
 // contains multiple edge nodes upgrade status.
 // +kubebuilder:validation:Type=object
 type NodeUpgradeJobStatus struct {
 	// State represents for the state phase of the NodeUpgradeJob.
 	// There are three possible state values: "", upgrading and completed.
-	State UpgradeState `json:"state,omitempty"`
+	State           api.State  `json:"state,omitempty"`
+	CurrentVersion  string     `json:"currentVersion,omitempty"`
+	HistoricVersion string     `json:"historicVersion,omitempty"`
+	Event           string     `json:"event,omitempty"`
+	Action          api.Action `json:"action,omitempty"`
+	Reason          string     `json:"reason,omitempty"`
+	Time            string     `json:"time,omitempty"`
```

Comment:
Does the `Time` indicate the latest update time?

Reply:
Yes. It is the event reporting time.